src/psyclone/psyir/backend/fortran.py

```diff
@@ -109,6 +109,79 @@ def gen_dims(symbol):
     return dims
 
 
+def gen_datatype(symbol):
+    '''Given a Symbol instance as input, return the datatype of the
+    symbol including any specific precision properties.
+
+    :param symbol: the symbol instance.
+    :type symbol: :py:class:`psyclone.psyGen.Symbol`
+
+    :returns: the Fortran representation of the symbol's datatype \
+    including any precision properties.
+    :rtype: str
+
+    '''
+    if symbol.datatype not in ["real", "integer", "character", "boolean"]:
+        raise NotImplementedError(
+            "unsupported datatype '{0}' found in gen_datatype().".
+            format(symbol.datatype))
+
+    if symbol.datatype == "boolean":
+        # boolean is the only datatype name that does not directly
+        # match the Fortran datatype name
+        datatype = "logical"
+    else:
+        datatype = symbol.datatype
+    if not symbol.precision:
+        # This symbol has no precision information so simply return
+        # the name of the datatype.
+        return datatype
+    if isinstance(symbol.precision, int):
+        if datatype not in ['real', 'integer', 'logical']:
+            raise VisitorError("Explicit precision not support for datatype "
+                               "'{0}' in Fortran backend.".format(datatype))
+        if datatype == 'real' and symbol.precision not in [4, 8, 16]:
+            raise VisitorError(
+                "Datatype 'real' supports fixed precision of [4, 8, 16] but "
+                "found '{0}'.".format(symbol.precision))
+        if datatype in ['integer', 'logical'] and symbol.precision not in \
+           [1, 2, 4, 8, 16]:
+            raise VisitorError(
+                "Datatype '{0}' supports fixed precision of [1, 2, 4, 8, "
+                "16] but found '{1}'.".format(datatype, symbol.precision))
+        # Precision has an an explicit size. Use the "type*size" Fortran
+        # extension for simplicity. We could have used
+        # type(kind=selected_int|real_kind(size)) or, for Fortran 2008,
+        # ISO_FORTRAN_ENV; type(type64) :: MyType.
+        return "{0}*{1}".format(datatype, symbol.precision)
+
+    if isinstance(symbol.precision, Symbol.Precision):
+        # The precision information is not absolute so is either
+        # machine specific or is specified via the compiler. Fortran
+        # only distinguishes relative precision for single and double
+        # precision reals.
+        if datatype.lower() == "real" and \
+           symbol.precision == Symbol.Precision.DOUBLE:
+            return "double precision"
+        import logging
```

Comment:
As discussed, although this is a clear use case for logging, we shouldn't just introduce it here without also adding top-level controls to the driver script. Therefore please comment it out for now and add a reference to the logging Issue (#11 ).

Reply:
(Also, you have superfluous parentheses in the `logging.warning` call.

Reply:
Oh, alright then. As agreed I'll keep the code for future reference, but comment it out.